### PR TITLE
Add jest testing

### PR DESCRIPTION
**Pull request by ndittren**
* Jest tests

### CODE REVIEW

**Review by sdreher — COMMENTED** [review]
Is this integrated into `make` yet? You might need to add the jstest target into the project `Makefile`.

I'm seeing a few errors when I run this via `npm test` for questionDashboard.test.tsx, levelsDashboard.test.tsx and exerciseSpace.test.tsx. Example output looks like so:

```
[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "ReferenceError: fetch is not defined".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
 FAIL  media/js/src/exerciseSpace.test.tsx
  ● Test suite failed to run

    Jest worker encountered 4 child process exceptions, exceeding retry limit

      at ChildProcessWorker.initialize (node_modules/jest-worker/build/workers/ChildProcessWorker.js:185:21)
```

**Comment by ndittren** [comment]
@sdreher I added the jstest as a make target and also added it to the `make all` target so that it runs here. The tests were passing fine on my machine but I see that a library is necessary for fetch.

File: Makefile [hunk 1]
```diff
@@ -34,6 +34,10 @@ cypress-test: js-build
 	npm run cypress:test
 .PHONY: cypress-test
 
+js-test: $(JS_SENTINAL)
```
Comment (sdreher) [inline Makefile:37]: Noting a duplication here - there is a `jstest` target in `js.mk` that does an `npm test`. Would that common target work? Or does it need to be `npm run test`?

**Review by sdreher — COMMENTED** [review]
Excellent - all passing on my env now. One note about the make target to address.

File: Makefile (hunk 1 above)
Comment (ndittren) [reply Makefile]: Whoops! Yes that's a duplicate. Removed.

**Pull request merged**